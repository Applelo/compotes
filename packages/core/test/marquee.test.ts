import { afterAll, beforeAll, describe, expect, it } from 'vitest'
import type { Browser } from 'playwright'
import { chromium } from 'playwright'

let browser: Browser

beforeAll(async () => {
  browser = await chromium.launch()
})

afterAll(async () => {
  browser.close()
})

describe('marquee', async () => {
  it.concurrent('events', async () => {
    const page = await browser.newPage()
    const events = [
      'c.marquee.play',
      'c.marquee.pause',
      'c.marquee.loop',
    ]
    const playEvent = page.waitForEvent('console', msg => msg.text().includes('c.marquee.play'))
    const pauseEvent = page.waitForEvent('console', msg => msg.text().includes('c.marquee.pause'))
    const loopEvent = page.waitForEvent('console', msg => msg.text().includes('c.marquee.loop'))
    await page.goto('http://localhost:3000/marquee.html')

    await page.evaluate(() => {
      const el = document.querySelector('.c-marquee')
      if (!el)
        return
      [
        'c.marquee.play',
        'c.marquee.pause',
        'c.marquee.loop',
      ].forEach((e) => {
        el.addEventListener(e, () => {
          // eslint-disable-next-line no-console
          console.log(e)
        })
      })
    })

    const playBtn = page.locator('.js-marquee-play').first()
    const pauseBtn = page.locator('.js-marquee-pause').first()
    await pauseBtn.click()
    await playBtn.click()

    const result = (await Promise.all([
      playEvent,
      pauseEvent,
      loopEvent,
    ])).map(item => item.text())

    events.forEach(e => expect(result).toContain(e))
  })
})
