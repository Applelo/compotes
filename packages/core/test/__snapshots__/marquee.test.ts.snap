// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`marquee > destroy 1`] = `
"<!DOCTYPE html><html lang="en"><head>
    <link rel="icon" href="data:image/svg+xml,<svg xmlns=%22http://www.w3.org/2000/svg%22 viewBox=%220 0 100 100%22><text y=%22.9em%22 font-size=%2290%22>🍯</text></svg>">
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Marquee</title>
    <script type="module" crossorigin="" src="/assets/marquee-XtOe3rLL.js"></script>
    <link rel="modulepreload" crossorigin="" href="/assets/_parent-PLeNLLls.js">
    <link rel="modulepreload" crossorigin="" href="/assets/index.esm-CjV0_wFH.js">
    <link rel="stylesheet" crossorigin="" href="/assets/marquee-ybAlwBfl.css">
  </head>
  <body>
    <div class="c-marquee" style="width: 200px;">
      <ul class="c-marquee-container">
        <li>Marquee Left Direction</li>
        <li><img width="25" height="25" src="https://vitejs.dev/logo.svg"></li>
      </ul>
    </div>
    <button class="js-marquee-play">
      Play
    </button>
    <button class="js-marquee-pause">
      Pause
    </button>
    <button class="js-destroy">
      Destroy
    </button>
  

</body></html>"
`;
